src/EPR.RegulatorService.Frontend.Web/Controllers/RegistrationSubmissions/RegistrationSubmissionsController.cs

```diff
@@ -165,23 +166,62 @@ public async Task<IActionResult> GrantRegistrationSubmission(Guid? organisationI
     {
         _currentSession = await _sessionManager.GetSessionAsync(HttpContext.Session);
 
-        if (!GetOrRejectProvidedOrganisationId(organisationId, out RegistrationSubmissionDetailsViewModel existingModel))
+        if (!GetOrRejectProvidedOrganisationId(organisationId, out var existingModel))
         {
             return RedirectToAction(PagePath.PageNotFound, "RegistrationSubmissions");
         }
 
-        SetBackLink(Url.RouteUrl("SubmissionDetails", new { organisationId }), false);
+        SetBackLink($"{PagePath.RegistrationSubmissionDetails}/{existingModel.OrganisationId}");
 
-        var model = new GrantRegistrationSubmissionViewModel()
+        var model = new GrantRegistrationSubmissionViewModel
         {
-            OrganisationId = organisationId.Value
+            OrganisationId = existingModel.OrganisationId
         };
 
         ViewBag.BackToAllSubmissionsUrl = Url.Action("RegistrationSubmissions");
 
         return View(nameof(GrantRegistrationSubmission), model);
     }
 
+    [HttpPost]
+    [Route(PagePath.GrantRegistrationSubmission + "/{organisationId:guid}", Name = "GrantRegistrationSubmission")]
+    public async Task<IActionResult> GrantRegistrationSubmission(GrantRegistrationSubmissionViewModel model)
+    {
+        _currentSession = await _sessionManager.GetSessionAsync(HttpContext.Session);
+
+        if (!GetOrRejectProvidedOrganisationId(model.OrganisationId, out var existingModel))
+        {
+            return RedirectToAction(PagePath.PageNotFound, "RegistrationSubmissions");
+        }
+
+        if (!ModelState.IsValid && model.IsGrantRegistrationConfirmed == null)
+        {
+            SetBackLink($"{PagePath.RegistrationSubmissionDetails}/{existingModel.OrganisationId}");
+            return View(nameof(GrantRegistrationSubmission), model);
+        }
+        else
+        {
+            if (model.IsGrantRegistrationConfirmed.Value)
```

Comment:
Remove the condition or react to it's failure - it is redundant at this point however...